opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/model/BaseIssueReferralModel.java

```diff
@@ -0,0 +1,137 @@
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public LiveData<List<ReferralServiceObject>> getReferralServicesList(List<String> referralServiceIds) {
+        try {
+            ReferralServiceRepository referralServiceRepository = new ReferralServiceRepository(ReferralLibrary.getInstance().getRepository());
+
+            List<ReferralServiceObject> servicesList = new ArrayList<>();
+            if (referralServiceIds != null) {
+                for (String serviceId : referralServiceIds) {
+                    try {
+                        servicesList.add(referralServiceRepository.getReferralServiceById(serviceId));
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }
+            } else {
+                servicesList = referralServiceRepository.getReferralServices();
+            }
+
+            MutableLiveData<List<ReferralServiceObject>> liveData = new MutableLiveData<>();
+            liveData.setValue(servicesList);
+
+            return liveData;
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public List<ReferralServiceIndicatorObject> getIndicatorsByServiceId(String serviceId) {
+        try {
+            ReferralServiceIndicatorRepository indicatorRepository = new ReferralServiceIndicatorRepository(ReferralLibrary.getInstance().getRepository());
+            return indicatorRepository.getServiceIndicatorsByServiceId(serviceId);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    @Override
+    public String mainSelect(String tableName, String mainCondition) {
+        SmartRegisterQueryBuilder queryBuilder = new SmartRegisterQueryBuilder();
+        queryBuilder.SelectInitiateMainTable(tableName, this.mainColumns(tableName));
+        return queryBuilder.mainCondition(mainCondition);
+    }
+
+    protected String[] mainColumns(String tableName) {
+        return new String[]{tableName + ".relationalid", tableName + "." + "base_entity_id", tableName + "." + "first_name", tableName + "." + "middle_name", tableName + "." + "last_name", tableName + "." + "unique_id", tableName + "." + "gender", tableName + "." + "dob", tableName + "." + "dod"};
+    }
+
+    @Override
+    public JSONObject getFormWithValuesAsJson(String formName, String entityId, String currentLocationId, MemberObject memberObject) throws Exception {
+        JSONObject jsonForm = JsonFormUtils.getFormAsJson(formName);
+        JsonFormUtils.getRegistrationForm(jsonForm, entityId, currentLocationId);
+
+        JSONObject formWithValues = setFormValues(jsonForm, memberObject);
+
+        return formWithValues;
+    }
+
+    @VisibleForTesting
+    public JSONObject setFormValues(JSONObject form, MemberObject memberObject) {
+        try {
+            JSONArray fieldsArray = form.getJSONObject("step1").getJSONArray("fields");
```

Comment:
Making this configurable can be helpful instead of hardcoding the step number. Use constant for *fields* its available via ```JsonFormConstants``` class